### ISSUE TITLE
Filter on connection causes error "lookup was already seen with a different queryset"

### ISSUE THREAD

**Issue by JanBednarik** [blb-ventures/strawberry-django-plus#47]
I have types and schema like:

```python
@gql.django.type(GroupMembership)
class GroupMembershipType(relay.Node):
    group: "GroupType"
    is_auto: gql.auto
    ...

@gql.django.filter(Group, lookups=True)
class GroupFilter:
    name: gql.auto

@gql.django.type(Group, filters=GroupFilter)
class GroupType(relay.Node):
    name: gql.auto
    groupmembership: Optional[List[GroupMembershipType]]
    ...

@strawberry.type
class Query:
    all_groups: relay.Connection[GroupType] = gql.django.connection()

schema = strawberry.Schema(query=Query, extensions=[DjangoOptimizerExtension])
```

When I run query like this everything is fine:

```graphql
{
  allGroups {
    edges {
      node {
        name
        groupmembership {
          isAuto
        }
      }
    }
  }
}
```

But when I use filters on allGroups like `allGroups (filters: {name: {startsWith: "foo"}})` it returns error:

```json
{
  "data": null,
  "errors": [
    {
      "message": "'groupmembership_set' lookup was already seen with a different queryset. You may need to adjust the ordering of your lookups.",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "allGroups"
      ]
    }
  ]
}
```

Versions:

```text
strawberry-django-plus==1.10.2
strawberry-graphql==0.111.2
strawberry-graphql-django==0.2.5
```

**Comment by darwinhace** [comment]
+1, same problem with me

```
@gql.django.filter(model=Shelf, lookups=True)
class ShelfFilter:
    rack: gql.auto
    bay: gql.auto
    row: gql.auto

@gql.django.type(model=models.Shelf, filters=ShelfFilter)
class Shelf(relay.Node):
    rack: Rack
    bay: gql.auto
    row: gql.auto
    bins: List["Bin"]


@gql.django.type(model=models.Bin)
class Bin(relay.Node):
    shelf: Shelf
    position: gql.auto
    size: gql.auto
    label: gql.auto
```
query:
```
{
  shelves(filters:{bay:{iContains:"A"}}) {
    edges {
      node {
        id
        rack {
          id
          name
        }
        bay
        row
        bins {
          id
          label
        }
      }
    }
  }
}
```

results:
```
{
  "data": null,
  "errors": [
    {
      "message": "'bins' lookup was already seen with a different queryset. You may need to adjust the ordering of your lookups.",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "shelves"
      ]
    }
  ]
}
```

**Comment by bellini666** [comment]
Thanks for reporting this. Will take a look at this very soon!

**Comment by bellini666** [comment]
@JanBednarik @darwinhace this should be fixed in the release 1.11 that I'm doing atm. Please reopen if it is not.

**Comment by JanBednarik** [comment]
It's working. Thank you.
